test/MinusTest.cpp

```diff
@@ -196,3 +263,221 @@ TEST(Minus, columnOriginatesFromGraphOrUndef) {
+
+    auto lazyResult = result.idTables();
+    auto it = lazyResult.begin();
+    ASSERT_NE(it, lazyResult.end());
+
+    EXPECT_EQ(it->idTable_, expected);
+
+    EXPECT_EQ(++it, lazyResult.end());
+  }
+}
+
+// _____________________________________________________________________________
+TEST(Minus, lazyMinusWithPermutedColumns) {
+  auto qec = ad_utility::testing::getQec();
+
+  auto expected = makeIdTableFromVector({{1, 11, 111}, {3, 33, 333}});
+
+  auto left = ad_utility::makeExecutionTree<ValuesForTesting>(
+      qec, makeIdTableFromVector({{1, 11, 111}, {2, 22, 222}, {3, 33, 333}}),
+      std::vector<std::optional<Variable>>{Variable{"?x"}, Variable{"?y"},
+                                           Variable{"?z"}},
+      false, std::vector<ColumnIndex>{2});
+  auto right = ad_utility::makeExecutionTree<ValuesForTesting>(
+      qec, makeIdTableFromVector({{2222, 222}}),
+      std::vector<std::optional<Variable>>{Variable{"?a"}, Variable{"?z"}},
+      false, std::vector<ColumnIndex>{1});
+  Minus minus{qec, left, right};
+
+  qec->getQueryTreeCache().clearAll();
+
+  auto result = minus.computeResultOnlyForTesting(true);
+
+  ASSERT_FALSE(result.isFullyMaterialized());
+
+  auto lazyResult = result.idTables();
+  auto it = lazyResult.begin();
+  ASSERT_NE(it, lazyResult.end());
+
+  EXPECT_EQ(it->idTable_, expected);
+
+  EXPECT_EQ(++it, lazyResult.end());
+}
+
+// _____________________________________________________________________________
+TEST(Minus, lazyMinusExceedingChunkSize) {
+  {
+    std::vector<IdTable> expected;
+    expected.push_back(makeIdTableFromVector({{Id::makeFromInt(3)}}));
+
+    std::vector<IdTable> leftTables;
+    leftTables.push_back(
+        makeIdTableFromVector({{1}, {2}, {3}}, &Id::makeFromInt));
+    std::vector<IdTable> rightTables;
+    rightTables.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(1)));
+    rightTables.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(2)));
+
+    testMinus(std::move(leftTables), std::move(rightTables),
+              std::move(expected), true);
+  }
+  {
+    std::vector<IdTable> expected;
+    expected.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(2)));
+
+    std::vector<IdTable> leftTables;
+    leftTables.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(1)));
+    leftTables.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(2)));
+    leftTables.push_back(createIdTableOfSizeWithValue(
+        qlever::joinHelpers::CHUNK_SIZE + 1, Id::makeFromInt(3)));
+    std::vector<IdTable> rightTables;
+    rightTables.push_back(makeIdTableFromVector({{1}, {3}}, &Id::makeFromInt));
+
+    testMinus(std::move(leftTables), std::move(rightTables),
+              std::move(expected), true);
```

Comment:
The somewhat correct merging of the local vocab can or could be tested. probably should.